internal/cmd/controller/gitops/reconciler/gitjob_controller.go

```diff
@@ -77,54 +77,54 @@ func (r *GitJobReconciler) SetupWithManager(mgr ctrl.Manager) error {
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.15.0/pkg/reconcile
 func (r *GitJobReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	logger := log.FromContext(ctx).WithName("gitjob")
-	var gitRepo v1alpha1.GitRepo
+	gitrepo := &v1alpha1.GitRepo{}
 
-	if err := r.Get(ctx, req.NamespacedName, &gitRepo); err != nil && !errors.IsNotFound(err) {
+	if err := r.Get(ctx, req.NamespacedName, gitrepo); err != nil && !errors.IsNotFound(err) {
 		return ctrl.Result{}, err
 	} else if errors.IsNotFound(err) {
 		logger.V(1).Info("Gitrepo deleted, cleaning up poll jobs")
 		r.GitPoller.CleanUpGitRepoPollJobs(ctx)
 		return ctrl.Result{}, nil
 	}
 
-	logger = logger.WithValues("generation", gitRepo.Generation, "commit", gitRepo.Status.Commit)
+	logger = logger.WithValues("generation", gitrepo.Generation, "commit", gitrepo.Status.Commit)
 	ctx = log.IntoContext(ctx, logger)
 
 	logger.V(1).Info("Reconciling GitRepo")
 
-	r.GitPoller.AddOrModifyGitRepoPollJob(ctx, gitRepo)
+	r.GitPoller.AddOrModifyGitRepoPollJob(ctx, *gitrepo)
 
 	var job batchv1.Job
 	err := r.Get(ctx, types.NamespacedName{
-		Namespace: gitRepo.Namespace,
-		Name:      jobName(&gitRepo),
+		Namespace: gitrepo.Namespace,
+		Name:      jobName(gitrepo),
 	}, &job)
 	if err != nil && !errors.IsNotFound(err) {
 		return ctrl.Result{}, fmt.Errorf("error retrieving git job: %w", err)
 	}
 
-	if errors.IsNotFound(err) && gitRepo.Status.Commit != "" {
-		if err := r.validateExternalSecretExist(ctx, &gitRepo); err != nil {
-			nsname := types.NamespacedName{Namespace: gitRepo.Namespace, Name: gitRepo.Name}
-			return ctrl.Result{}, grutil.UpdateErrorStatus(ctx, r.Client, nsname, gitRepo.Status, err)
+	if errors.IsNotFound(err) && gitrepo.Status.Commit != "" {
+		if err := r.validateExternalSecretExist(ctx, gitrepo); err != nil {
+			nsname := types.NamespacedName{Namespace: gitrepo.Namespace, Name: gitrepo.Name}
```

Comment:
I think you could use `client.ObjectKeyFromObject(gitrepo)` here.

https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/client#ObjectKeyFromObject

Reply:
Interesting function, but here we can just use the existing `req.NamespacedName`.